File: src/dependent/payments/receipt.py
from decimal import Decimal


class Receipt:
    def __init__(self, payment_id: str, reference: str, total: Decimal) -> None:
        self._payment_id = payment_id
        self._reference = reference
        self._total = total

    @property
    def payment_id(self) -> str:
        return self._payment_id

    @property
    def reference(self) -> str:
        return self._reference

    @property
    def total(self) -> Decimal:
        return self._total

File: src/dependent/payments/Receipt.java
package org.example.dependent.payments;

import java.math.BigDecimal;

public class Receipt {
    private final String paymentId;
    private final String reference;
    private final BigDecimal total;

    public Receipt(String paymentId, String reference, BigDecimal total) {
        this.paymentId = paymentId;
        this.reference = reference;
        this.total = total;
    }

    public String getPaymentId() {
        return paymentId;
    }

    public String getReference() {
        return reference;
    }

    public BigDecimal getTotal() {
        return total;
    }
}


